ToolBox/EstimizeDataDownloader/EstimizeDataDownloader.cs

```diff
@@ -0,0 +1,133 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Configuration;
+using QuantConnect.Data;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace QuantConnect.ToolBox.EstimizeDataDownloader
+{
+    public abstract class EstimizeDataDownloader : IDataDownloader
+    {
+        private readonly string _clientKey;
+
+        /// <summary>
+        /// Control the rate of download per unit of time.
+        /// </summary>
+        public RateGate IndexGate { get; }
+
+        protected EstimizeDataDownloader()
+        {
+            _clientKey = Config.Get("estimize-economics-auth-token");
+
+            // Represents rate limits of 10 requests per 1.1 second
+            IndexGate = new RateGate(10, TimeSpan.FromSeconds(1.1));
```

Comment:
Interesting rate limit, lol, 10 requests per 1100 milliseconds? First time for everything!

Reply:
It was inspired by the @gsalaz98 ' https://github.com/QuantConnect/Lean/pull/3266 PR. :-D